controllers/managedocs_controller.go

```diff
@@ -130,6 +137,7 @@ type ManagedOCSReconciler struct {
 	cephIngressNetworkPolicy           *netv1.NetworkPolicy
 	providerAPIServerNetworkPolicy     *netv1.NetworkPolicy
 	prometheus                         *promv1.Prometheus
+	prometheusService                  *corev1.Service
```

Comment:
As this is the first owned resource of type `Service` we are reconciling, you need to set up an owner watch on this type at the `SetupWithManager` function. It should look something like this:
```Owns(&orev1.Service{})```

Reply:
Done